### PR TITLE
feat(cli): Add recce-cloud doctor command for environment diagnostics (DRC-2592)

### PR DESCRIPTION
**Pull request by even-wei**
## Summary
- Add `recce-cloud doctor` command for environment diagnostics (DRC-2592)
- Refactor doctor command using Service Layer pattern for better maintainability
- Fix `--type prod` upload to work outside CI environment
- Fix production metadata check to use correct API field (`adapter_type`)
- Address code review feedback with improved error handling and tests

### New `recce-cloud doctor` Command
Validates Recce Cloud setup and provides actionable suggestions:
1. **Login Status** - Checks authentication
2. **Project Binding** - Checks `recce-cloud init` configuration
3. **Production Metadata** - Checks if production session has data
4. **Dev Session** - Checks for development sessions

```
$ recce-cloud doctor

╭────────────────────────────────────────╮
│   🩺 Recce Doctor                      │
│   Checking your Recce Cloud setup...   │
╰────────────────────────────────────────╯

1. Login Status ($ recce-cloud login)
✓ Logged in as user@example.com

2. Project Binding ($ recce-cloud init)
✓ Bound to myorg/myproject

3. Production Metadata
✓ Found production session "prod" (uploaded 2h ago)

4. Dev Session
✓ Found dev session "feature-branch" (uploaded 1h ago)

✓ All checks passed! Your Recce setup is ready.

Next step:
  $ recce-cloud review --session-name <session_name>
```

Supports `--json` flag for scripting (now includes `suggestion` field).

### Code Refactoring
- Extracted business logic to `services/diagnostic_service.py`
- Extracted CLI presentation to `commands/diagnostics.py`
- Reduced `cli.py` from 1,557 to 1,252 lines
- Establishes pattern for future command refactoring (see DRC-2604)

### Bug Fixes
- `recce-cloud upload --type prod` now works outside CI environment by auto-fetching production session ID
- Production metadata check correctly uses `adapter_type` field from API
- Fixed guidance message to suggest `upload --type prod` instead of `init`

### Code Review Feedback (Latest Commit)
- Changed next step suggestion from `summarize` to `review --session-name`
- Added `suggestion` field to JSON output for scripting use cases
- Added None guards for `org_id`, `project_id`, and `session_id` to handle API edge cases
- Added `logger.exception()` for traceback logging in exception handlers
- Added comprehensive tests for `_get_production_session_id()` and API error paths

## Test plan
- [x] 9 unit tests for doctor command (all passing)
- [x] 9 unit tests for `_get_production_session_id()` error paths
- [x] 2 unit tests for API error handling in diagnostic service
- [x] 1 test for JSON output including suggestion field
- [x] Existing upload tests still pass
- [x] Manual testing of `recce-cloud doctor` command
- [x] Manual testing of `recce-cloud upload --type prod` outside CI

## Follow-up
- [ ] Add `recce-cloud review --session-name` command (separate issue)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: recce_cloud/recce_cloud/cli.py [hunk 1]
```diff
@@ -596,19 +679,38 @@ def upload(target_path, session_id, session_name, skip_confirmation, cr, session
         # Platform-specific workflow: Use platform APIs to create session and upload
```
Comment (wcchang1115) [inline recce_cloud/recce_cloud/cli.py]: I feel the comment would mislead that the else block is for platform-specific workflow.

File: recce_cloud/recce_cloud/cli.py [hunk 2]
```diff
@@ -359,10 +365,87 @@ def init(org, project, status, clear):
         console.print(f"[red]Error:[/red] Failed to fetch data from Recce Cloud: {e}")
         sys.exit(1)
     except Exception as e:
+        logger.exception("Unexpected error during init: %s", e)
```
Comment (wcchang1115) [inline recce_cloud/recce_cloud/cli.py]: I saw some `logger.exception` are introduced in this PR.
Is it intended to print the exception stack to the terminal?

File: recce_cloud/recce_cloud/services/diagnostic_service.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        self._token = os.getenv("RECCE_API_TOKEN") or get_api_token()
+
+        if not self._token:
+            return CheckResult(
+                status=CheckStatus.FAIL,
+                message="Not logged in",
+                suggestion="Run 'recce-cloud login' to authenticate",
+            )
+
+        is_logged_in, email = check_login_status()
```
Comment (wcchang1115) [inline recce_cloud/recce_cloud/services/diagnostic_service.py:154]: `check_login_status` get token from the recce's profile only. should consider the env variable case.

**Review by wcchang1115 — CHANGES_REQUESTED** [review]
Hi even,
There are a few questions for you to further check.
